projectional/src/main/java/jetbrains/jetpad/projectional/selection/SelectionListener.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2012-2016 JetBrains s.r.o
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package jetbrains.jetpad.projectional.selection;
+
+public interface SelectionListener {
+  void onSelectionOpened(long selectionId, Selection selection);
```

Comment:
It is needed to uniquely identify selection regions. Particularly if resources are allocated per selection region they need to be disposed.